modules/standard/FileSystem.chpl

```diff
@@ -1205,8 +1205,19 @@ iter listdir(path: string = ".", hidden: bool = false, dirs: bool = true,
     }
     closedir(dir);
   } else {
+    /* perror() is used here because it asks the C stdlib to print out a
+       message specific to the current value of errno */
     extern proc perror(s: c_string);
-    perror("error in listdir(): ");
+
+    // stderr.writef would be cleaner than this try/catch block, but it throws
+    var errorMsg: string;
+    try {
+      errorMsg = "error in listdir() with argument '%s'".format(path);
+    } catch e {
+      // Generic error message in case format fails
+      perror("Error in listdir(): ");
```

Comment:
It seems unlikely that C's `errno` would still be set properly by the time we've run through format + try/catch code paths in Chapel.  Isn't it only valid immediately after the operation that set it?  Seems like once we start introducing more complex code here, we should grab errno immediately and then process it within Chapel directly.

Reply:
Good point. I chatted with @lydia-duncan, and believe the right solution involves wrapping `opendir()` with `chpl_fs_opendir()` that returns a `syserr`, like we do with other FS calls in `runtime/src/chpl-file-utils.c`.